datadog/provider.go

```diff
@@ -91,16 +97,40 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	ok, err := communityClient.Validate()
 	if err != nil {
 		log.Printf("[ERROR] Datadog Client validation error: %v", err)
-		return communityClient, err
+		return nil, err
 	} else if !ok {
 		err := errors.New(`Invalid or missing credentials provided to the Datadog Provider. Please confirm your API and APP keys are valid and are for the correct region, see https://www.terraform.io/docs/providers/datadog/ for more information on providing credentials for the Datadog Provider`)
 		log.Printf("[ERROR] Datadog Client validation error: %v", err)
-		return communityClient, err
+		return nil, err
 	}
 	log.Printf("[INFO] Datadog Client successfully validated.")
 
+	// Initialize the official Datadog client
+	auth := context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: d.Get("api_key").(string),
+			},
+			"appKeyAuth": {
+				Key: d.Get("app_key").(string),
+			},
+		},
+	)
+	config := datadog.NewConfiguration()
+	if apiURL := d.Get("api_url").(string); apiURL != "" {
+		if strings.Contains(apiURL, "datadoghq.eu") {
+			auth = context.WithValue(auth, datadog.ContextServerVariables, map[string]string{
+				"site": "datadoghq.eu",
+			})
+		}
+	}
```

Comment:
I think we need to be smarter here, in case someone is not sending to a datadog URL directly